File: src/app/auth/auth.config.ts
import { Injectable, } from '@nestjs/common';
import { ConfigService, } from '@nestjs/config';

interface JwtTokenInterface {
  secret: string;
}

export interface JwtAuthInterface {
  file: JwtTokenInterface;
  user: JwtTokenInterface;
  admin: JwtTokenInterface;
}

@Injectable()
export class AuthConfig {
  /** JWT auth secrets */
  public jwt: JwtAuthInterface;

  constructor(private readonly configService: ConfigService) {
    this.jwt = {
      file: {
        secret: this.configService.getOrThrow<string>('FA_SECRET'),
      },
      user: {
        secret: this.configService.getOrThrow<string>('UA_SECRET'),
      },
      admin: {
        secret: this.configService.getOrThrow<string>('AA_SECRET'),
      },
    };
  }
}

File: app/auth/auth.config.go
package auth

import (
	"file-service/app/utils"

	"log"
)

type JWTConfig struct {
	Secret string
}

type Config map[TokenType]JWTConfig

var (
	AuthConfig Config
)

func init() {
	log.Print("Load auth config")
	AuthConfig = make(Config)
	AuthConfig[TokenTypeAdminAccess] = JWTConfig{
		Secret: utils.GetEnv("AA_SECRET", ""),
	}
	AuthConfig[TokenTypeUserAccess] = JWTConfig{
		Secret: utils.GetEnv("UA_SECRET", ""),
	}
	AuthConfig[TokenTypeFileAccess] = JWTConfig{
		Secret: utils.GetEnv("FA_SECRET", ""),
	}
}
